crates/polars-expr/src/expressions/group_iter.rs

```diff
@@ -75,7 +75,9 @@ impl AggregationContext<'_> {
 
 impl AggregationContext<'_> {
     /// Iterate over groups lazily, i.e., without greedy aggregation into an AggList.
-    pub(super) fn iter_groups_lazy(
+    ///
+    /// # Safety: AmortSeries does not live longer than the next call to `next`.
+    pub(super) unsafe fn iter_groups_lazy(
```

Comment:
Since you made this unsafe and you only had to change the `ApplyExpr` usage I think we can conclude this is only used there. Can we rewrite this so it doesn't involve an unsafe function returning an `AmortSeries` which is then immediately deep cloned, and instead it just uses some safe form of iteration which does any clones ahead of time if necessary?